File: HahnLeaveAbsenceManagementFrontend/src/app/core/interfaces/auth-interfaces/login-response.ts
import {UserRole} from "../user-interfaces/user-Role";

export interface LoginResponse {
  id: string;
  email: string;
  firstName: string;
  lastName: string;
  role: UserRole;
  leavesLeft: number;
  token: string ;
  tokenExpiresAt: Date;
}

File: HahnLeaveAbsenceManagementBackend/HahnLeaveAbsenceManagement.Application/User/Models/LoginResponse.cs
using HahnLeaveAbsenceManagement.Domain.User;

namespace HahnLeaveAbsenceManagement.Application.User.Models;

public record LoginResponse()
{
    public Guid Id {get; set;}
    public string Email { get; set; }
    public string FirstName {get; set;}
    public string LastName {get; set;}
    public UserRole Role {get; set;}
    public int LeavesLeft { get; set; }
    public string? Token {get; set;}
    public DateTime? TokenExpiresAt {get; set;}
};